tests/utils/admincli/vmgroupmgmt.go

```diff
@@ -84,3 +91,56 @@ func ConfigRemove(ip string) (string, error) {
 	log.Printf("Removing the SingleNode Config DB on esx [%s] \n", ip)
 	return ssh.InvokeCommand(ip, admincli.RemoveLocalConfigDb)
 }
+
+// CheckVmgroupAvailability - checks for a vmgroup in list of vmgroups
+func CheckVmgroupAvailability(esxIP, vmgroupName string) bool {
+	log.Printf("Checking if vmgroup [%s] exists from esx [%s] \n", vmgroupName, esxIP)
+	cmd := admincli.ListVMgroups + "2>/dev/null | grep " + vmgroupName
+	out, _ := ssh.InvokeCommand(esxIP, cmd)
+	return strings.Contains(out, vmgroupName)
+}
+
+// CheckVolumeExistInVmgroup - verify volume's vmgroup is same as expected
+func CheckVolumeExistInVmgroup(esxIP, volName, vmgroupName string) bool {
+	log.Printf("Checking if volume [%s] belongs to vmgroup [%s] from esx [%s] \n", volName, vmgroupName, esxIP)
+	cmd := admincli.ListVolumes + " -c volume,vmgroup 2>/dev/null | grep " + volName
+	out, _ := ssh.InvokeCommand(esxIP, cmd)
+	return strings.Contains(out, vmgroupName)
+}
+
+// AddDatastoreToVmgroup - Grants datastore to a vmgroup whose access is controlled
+func AddDatastoreToVmgroup(ip, name, datastore string) (string, error) {
+	log.Printf("Adding datastore %s to  vmgroup %s,  on esx [%s] \n", datastore, name, ip)
+	cmd := admincli.AddDatastoreToVMgroup + name + " --datastore=" + datastore
+	log.Printf("CMD to add ds %s to vmgroup %s is : %s", datastore, name, cmd)
+	return ssh.InvokeCommand(ip, admincli.AddDatastoreToVMgroup+name+" --datastore="+datastore)
+}
+
+// IsDSAccessibleForVMgroup - Verifies if vmgroup has access rights to a datastore
+func IsDSAccessibleForVMgroup(ip, name, datastore string) bool {
+	log.Printf("Getting vmgroup %s access rights to datastore %s ,  on esx [%s] \n", name, datastore, ip)
+	cmd := admincli.GetAccessForVMgroup + name + " 2>/dev/null | awk -v OFS='\t' '{print $1, $2}' | grep " + datastore
+	out, _ := ssh.InvokeCommand(ip, cmd)
+	accessRights := strings.Fields(out)
```

Comment:
don't we have to check `out` is not empty? What happens when datastore access rule is not set for the vmgroup?